tests/test_multiprocess.py

```diff
@@ -422,3 +426,83 @@ def tearDown(self):
         del os.environ['prometheus_multiproc_dir']
         shutil.rmtree(self.tempdir)
         values.ValueClass = MutexValue
+
+
+class TestInMemoryCollector(unittest.TestCase):
+    def setUp(self):
+        self.tempdir = tempfile.mkdtemp()
+        os.environ['prometheus_multiproc_dir'] = self.tempdir
+        values.ValueClass = MultiProcessValue(lambda: 123)
+        self.registry = CollectorRegistry()
+        self.collector = InMemoryCollector(self.registry)
+
+    def tearDown(self):
+        del os.environ['prometheus_multiproc_dir']
+        shutil.rmtree(self.tempdir)
+        values.ValueClass = MutexValue
+        prometheus_client.multiprocess._metrics_cache = prometheus_client.multiprocess.MetricsCache()
+
+    def test_serves_empty_metrics_if_no_metrics_written(self):
+        self.assertEqual(self.collector.collect(), [])
+
+    def test_serves_empty_metrics_if_not_processed(self):
+        c1 = Counter('c', 'help', registry=None)
```

Comment:
The concept of a registry and collector is defined in the prometheus standard https://prometheus.io/docs/instrumenting/writing_clientlibs/. One quirk of this library, even before the fork, is that due to the file-based collection method, you can't have multiple CollectorRegistr(ies), sort of going against the standard

This means that regardless of what the registry is set to, it's still collected by the cleanup thread